### PR TITLE
Remove mediator_type and cleanup cpl_scalars

### PR DESCRIPTION
**Pull request by BinLiu-NOAA**
Remove mediator_type and cleanup cpl_scalars

### CODE REVIEW

File: NUOPC/hycom_nuopc_flags.F90 [hunk 1]
```diff
@@ -34,7 +34,18 @@ module hycom_nuopc_flags
     IMPORT_ERROR     = import_flag(-1), &
```
Comment (uturuncoglu) [inline NUOPC/hycom_nuopc_flags.F90]: I think it would be better to put comment/explanation about each of these new options.

**Review by uturuncoglu — COMMENTED** [review]
did we tested these changes by adding cpl_scalars to the list of export fields and try to write mediator history and restart files. If it works all the variables sent/received by ocean component will be 2d rather than 1d.

File: NUOPC/hycom_nuopc_flags.F90 (hunk 1 above)
Comment (danrosen25) [reply NUOPC/hycom_nuopc_flags.F90]: Done

**Comment by danrosen25** [comment]
@BinLiu-NOAA 
I rebased these changes onto the fix/radfl branch instead of merging the two branches.

@uturuncoglu 
I did test turning on cpl_scalars using ScalarFieldCount = 2 and it worked fine but I don't have the expertise to work with the mediator history and restart files. Please run the tests that you need to accomplish this. We're hoping to merge this PR as soon as possible.

**Review by danrosen25 — APPROVED** [review]
My approval is contingent on @uturuncoglu approval of changes related to coupling DATM and exporting cpl_scalars field.

**Comment by uturuncoglu** [comment]
@danrosen25 thanks. I think we could merge this and them I could start working on cdeps branch and test the changes. I also need to update regression tests also. @BinLiu-NOAA if you don't mind could you update me about the sync. Is it ready to work on CDEPS branch. Thanks.

**Review by uturuncoglu — APPROVED** [review]

**Comment by BinLiu-NOAA** [comment]
Thanks, @danrosen25 and @uturuncoglu! Merging this PR.  BTW, @uturuncoglu I will update you through email for the status of the ufs-weather-model level sync.

**Pull request merged**